tsdef/LocalParticipant.d.ts

```diff
@@ -15,7 +15,7 @@ export class LocalParticipant extends Participant {
   videoTracks: Map<Track.SID, LocalVideoTrackPublication>;
   signalingRegion: string;
 
-  publishTrack(track: LocalTrack | MediaStreamTrack, options?: LocalTrackPublishOptions): Promise<LocalTrackPublication>;
+  publishTrack(track: LocalTrack | MediaStreamTrack, options?: LocalTrackPublishOptions | MediaStreamTrackPublishOptions): Promise<LocalTrackPublication>;
```

Comment:
Instead of:
```
publishTrack(track: LocalTrack | MediaStreamTrack, options?: LocalTrackPublishOptions | MediaStreamTrackPublishOptions): Promise<LocalTrackPublication>;
```

Should it be?:
```
publishTrack(track: LocalTrack, options?: LocalTrackPublishOptions ): Promise<LocalTrackPublication>;
publishTrack(track: MediaStreamTrack, options?: MediaStreamTrackPublishOptions): Promise<LocalTrackPublication>;
```


With the second snippet - you can only pass `MediaStreamTrackPublishOptions` when publishing a `MediaStreamTrack`, and you can only pass `LocalTrackPublishOptions`, when publishing a `LocalTrack`.

The first snippet allows users to pass a `MediaStreamTrackPublishOptions` object when publishing a `LocalTrack`, and to pass a `LocalTrackPublishOptions` object when publishing a `MediaStreamTrack`. I don't think this should be allowed. I think the second snippet above is more consistent with our documentation. 

What do you think?

Reply:
Thanks Tim, I think that's a great idea and it makes total sense. I will make those changes now :)